synapseclient/client.py

```diff
@@ -917,7 +917,8 @@ def printEntity(self, entity, ensure_ascii=True) -> None:
         """
 
         if utils.is_synapse_id_str(entity):
-            entity = self._getEntity(entity)
+            synid, _ = utils.get_synid_and_version(entity)
+            entity = self._getEntity(synid)
```

Comment:
You would need to pass in the version here as `_getEntity` supports version:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/SYNPY-1398-support-synid-version/synapseclient/client.py#L5487